source/common/upstream/strict_dns_cluster.h

```diff
@@ -41,6 +41,10 @@ class StrictDnsClusterImpl : public BaseDynamicClusterImpl {
     const uint32_t port_;
     const Event::TimerPtr resolve_timer_;
     HostVector hosts_;
+
+    // All host map for current resolve target. When we have multiple resolve targets, multiple
+    // targets may contain two different host objects with the same address. This host map cannot be
+    // replaced by the read only all host map in the priority set.
     HostMap all_hosts_;
```

Comment:
The shared map is still worked for strict DNS clusters. However, we cannot use shared map to replace this `all_hosts_`.

Because strict DNS cluster will remove duplicate hosts in the resolve target. But if multiple resolve targets contain hosts with the same address, these hosts will exist at the same time. As a global host map, shared map cannot reserve two different hosts with the same address for two different resolve targets, nor can it keep the logic of strict DNS cluster unchanged.
So shared map cannot replace all_hosts_ (note that all_hosts_ here is a member of resolve target).

Reply:
I guess my concern would be around people trying to implement sticky LB for strict DNS but running into weird edge cases: suppose they use regular LB to select an endpoint from the STRICT_DNS cluster and wants to pin this lb selection

This works fine if there is only one host per ip within the cluster, but if this is configured with multiple resolve targets that resolve to the same ip, some of which have different `metadata`, you might run into a situation where the first LB selects a different endpoint than what the shared map gives you when you ask for the IP

I don't think we need to fix this, but perhaps some documentation around this limitation with the strict dns cluster would be good?

Reply:
> This works fine if there is only one host per ip within the cluster, but if this is configured with multiple resolve targets that resolve to the same ip, some of which have different metadata, you might run into a situation where the first LB selects a different endpoint than what the shared map gives you when you ask for the IP

Yes, this is a known limitation. I will add some new notes to illustrate this problem.